pkg/network/render.go

```diff
@@ -888,6 +905,49 @@ func renderIPTablesAlerter(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.
 	return manifests, nil
 }
 
+// renderNetworkingConsolePlugin renders the common objects related to the networking console plugin resources
+func renderNetworkingConsolePlugin(manifestDir string, bootstrapResult *bootstrap.BootstrapResult) ([]*uns.Unstructured, error) {
+	if !bootstrapResult.Infra.ConsolePluginCRDExists {
+		log.Printf("consoleplugins.console.openshift.io CRD does not exist yet")
+		return nil, nil
+	}
+
+	data := render.MakeRenderData()
+	data.Data["ReleaseVersion"] = os.Getenv("RELEASE_VERSION")
+
+	consolePluginImage, ok := os.LookupEnv("NETWORKING_CONSOLE_PLUGIN_IMAGE")
+	if !ok {
+		// temporary fallback until the image quay.io/openshift/origin-networking-console-plugin:latest
+		// will be published
+		tmpFallbackConsoleImage := "quay.io/orenc/networking-console-plugin:4.16"
+		data.Data["NetworkingConsolePluginImage"] = tmpFallbackConsoleImage
+	}
+	data.Data["NetworkingConsolePluginImage"] = consolePluginImage
+
+	manifests, err := render.RenderDir(filepath.Join(manifestDir, "networking-console-plugin"), &data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render networking-console-plugin manifests")
+	}
+
+	return manifests, nil
+}
+
+// registerNetworkingConsolePlugin enables console plugin for networking-console if not already enabled
+func registerNetworkingConsolePlugin(bootstrapResult *bootstrap.BootstrapResult, cl cnoclient.Client) error {
+	if !bootstrapResult.Infra.ConsolePluginCRDExists {
+		return nil
+	}
+	pluginName := "networking-consoleOperator-plugin"
+	consoleOperator := applyoperv1.Console(names.OPERATOR_CONFIG)
```

Comment:
`consoleOperator.Spec` is nil after this line, try this instead:
```suggestion
	consoleOperator := applyoperv1.Console(names.OPERATOR_CONFIG).WithSpec(applyoperv1.ConsoleSpec().WithPlugins(pluginName))
```

Reply:
I did it (with the addition of the required `FieldManager` in `ApplyOptions{}`, but i got a conflict error:
```
I0805 11:37:01.006859 1 log.go:245] Failed to render: registering networking-console-plugin "networking-console-plugin" with ConsoleOperator "cluster" has failed: Apply failed with 1 conflict: conflict with "operator" using operator.openshift.io/v1: .spec.plugins
```
could you please advise how can we avoid the conflict?

Reply:
You might need to set `ApplyOptions.Force` to true.

Reply:
ok, i've managed to get it working.
since `Plugins` is a list, the console object must be fetched before we attempt an apply operation, otherwise it will overwrite the list and remove the existing plugins.

Reply:
ugh :( The `Spec.Plugins` doesn't have a listType configured which means it is atomic. This doesn't make much sense in a scenario where multiple operators are supposed to edit it but that's the legacy API we need to handle.

To avoid racing and overriding others changes lets go back to a form of your initial approach. Something like this should do it:
```
func registerNetworkingConsolePlugin(bootstrapResult *bootstrap.BootstrapResult, cl cnoclient.Client) error {
	if !bootstrapResult.Infra.ConsolePluginCRDExists {
		return nil
	}

	console, err := cl.Default().OpenshiftOperatorClient().OperatorV1().Consoles().Get(context.TODO(), "cluster", metav1.GetOptions{})
	if err != nil {
		return errors.Wrap(err, "Failed to get Console Operator resource")
	}

	if sets.NewString(console.Spec.Plugins...).Has(pluginName) {
		return nil
	}
	console.Spec.Plugins = append(console.Spec.Plugins, pluginName)

	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
		_, err := cl.Default().OpenshiftOperatorClient().OperatorV1().Consoles().Update(context.TODO(), console, metav1.UpdateOptions{})
		return err
	})
}

```
P.S make pluginName a global const.

Reply:
changed the function back to use `Update()` instead of `Patch()` :+1: